upload/src/addons/SV/CustomFieldPerms/XF/Pub/Controller/Common.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace SV\CustomFieldPerms\XF\Pub\Controller;
+
+class Common {
+
+	/**
+	 * Insert additionalFilters into a reply.
+	 * These are then passed in to the custom_fields_macro via a template modification.
+	 *
+	 * @param $reply
+	 * @param $key
+	 */
+	public static function applyUsergroupCustomFieldPermissionFilters(&$reply, $key, $supplementaryFilters=[]) {

```

Comment:
My preference is this should be an XF2 repository so it can be override or extended. Since a repository lasts the life of a request with a single instance, there isn't any real overhead.